src/tests/unit/teams/sync-team.f90

```diff
@@ -24,6 +25,7 @@ program main
       end if
```

Comment:
This test isn't really actually testing the functionality of form team and sync team, it's more of a smoke test. It would be nice to add some assertions using get_team and team_number to verify this is functioning as expected.

Reply:
Also, some check on the synchornization. I also do not understand the point of the second set of calls to:

```
         sync team(team(CHILD_TEAM))
         sync team(team(CURRENT_TEAM))
```

Shouldn't calling each one once be sufficient?

Reply:
Well, I don't know what the initial author had in mind with that. I can only speculate, that he encountered some deadlocks when waiting for different teams.

I have reverted my changes to the sync_team.f90 and added two new tests with hopefully more "beef". Thank you for making me do this, because test_teams_1.f90 had me fix some errors in gfortran and in OpenCoarrays. Therefore there is a longer commit list now.